packages/app-builder-lib/src/winPackager.ts

```diff
@@ -392,9 +392,18 @@ export class WinPackager extends PlatformPackager<WindowsConfiguration> {
       return
     }
 
-    const outResourcesDir = path.join(packContext.appOutDir, "resources", "app.asar.unpacked")
-    // noinspection JSUnusedLocalSymbols
-    const fileToSign = await walk(outResourcesDir, (file, stat) => stat.isDirectory() || file.endsWith(".exe") || (this.isSignDlls() && file.endsWith(".dll")))
-    await BluebirdPromise.map(fileToSign, file => this.sign(file), { concurrency: 4 })
+    const directoriesToBeSigned = [["resources", "app.asar.unpacked"], ["swiftshader"]]
+
+    const outDirectories = directoriesToBeSigned.map(directoryPath => {
+      return path.join.apply(this, [packContext.appOutDir, ...directoryPath])
+    })
+
+    const filesToSignPromise = outDirectories.map(async outDir => {
+      return await walk(outDir, (file, stat) => stat.isDirectory() || file.endsWith(".exe") || (this.isSignDlls() && file.endsWith(".dll")))
+    })
+
+    const filesToSign = await Promise.all(filesToSignPromise)
+    const filesToSignFlat = filesToSign.flat(1)
+    await BluebirdPromise.map(filesToSignFlat, file => this.sign(file), { concurrency: 4 })
```

Comment:
Can we make this entire section more simple?
```
const signPromise = (filepath: string[]) => {
  const outDir = path.join(packContext.appOurDir, ...filepath)
  return walk(outDir, (file, stat) => stat.isDirectory() || file.endsWith(".exe") || (this.isSignDlls() && file.endsWith(".dll")))
}
const filesToSign = await Promise.all([
    signPromise(["resources", "app.asar.unpacked"]),
    signPromise(["swiftshader"])
])
await BluebirdPromise.map(filesToSign.flat(1), file => this.sign(file), { concurrency: 4 })
```

Reply:
Thanks for checking it @mmaietta, I updated it with a fixup commit [4611e1b](https://github.com/electron-userland/electron-builder/pull/6682/commits/4611e1be484da49124eeed232363eba72c3c5250) let me know if you think it is good and I can rebase it so there is only one commit for that.